currency-rates/internal/services/mail.go

```diff
@@ -1,42 +1,87 @@
 package services
 
 import (
+	"crypto/tls"
 	"net/smtp"
 
 	"go.uber.org/zap"
 )
 
+const serverName = "smtp.gmail.com"
 type Mail struct {
 	from string
 	pass string
+	client   *smtp.Client
 	l *zap.SugaredLogger
 }
 
-func NewEmail(from string, pass string, l *zap.SugaredLogger) *Mail {
+func NewEmail(from string, pass string, l *zap.SugaredLogger) (*Mail, error) {
+	client, err := smtp.Dial("smtp.gmail.com:587")
+	if err != nil {
+		return nil, err
+	}
+
+	tlsconfig := &tls.Config{
+		ServerName: serverName,
+	}
+	
+	if err = client.StartTLS(tlsconfig); err != nil {
+		return nil, err
+	}
+
+	auth := smtp.PlainAuth("", from, pass, serverName)
+	if err = client.Auth(auth); err != nil {
+		return nil, err
+	}
+	
 	return &Mail{
 		from: from,
 		pass: pass,
+		client: client,
 		l: l,
-	}
+	}, nil
 }
 
 func (m Mail) Send(to, body string) {
-	pass := m.pass
+	if m.client == nil {
+		m.l.Infof("SMTP client is not connected")
+		return
+	}
+
 	from := m.from
 
 	msg := "From: " + from + "\n" +
 		"To: " + to + "\n" +
 		"Subject: Dollar Rate\n\n" +
 		body
```

Comment:
This way of way strings concatenation has poor efficiency in Golang. Better use `fmt.Sprintf` here.

Reply:
Done